espnet2/tasks/diar.py

```diff
@@ -74,6 +82,15 @@
     type_check=AbsDecoder,
     default="linear",
 )
+attractor_choices = ClassChoices(
+    "attractor",
+    classes=dict(
+        rnn=RnnAttractor,
+        rnn_nograd=RnnAttractorNoGrad,
```

Comment:
I did not quite understand where this will be used. Could you show me how it is functioning? If you just want to freeze the params when training sometimes, you can just forward with `no_grad()`

Reply:
`rnn_nograd` corresponds to equations (23) and (24) in https://arxiv.org/pdf/2106.10654.pdf. My intension of adding this option is to easily use it for future experiments. However, if there is a better way to do this, I will be glad to fix it!

Reply:
I think you can add an option to the `RnnAttractor` class to enable no_grad by https://pytorch.org/docs/stable/generated/torch.no_grad.html Because at least for the structure, it's exact the same right?

Reply:
Thanks for the suggestion! I totally agree. Added an option to `RnnAttractor`.